src/init/features/hosting/github.ts

```diff
@@ -284,13 +286,15 @@ function writeChannelActionYMLFile(
   ymlPath: string,
   secretName: string,
   projectId: string,
+  repo: string,
   script?: string
 ): void {
   const workflowConfig: GitHubWorkflowConfig = {
     name: "Deploy to Firebase Hosting on PR",
     on: "pull_request",
     jobs: {
       ["build_and_preview"]: {
+        if: `github.repository == "${repo}"`, // secrets aren't accessible on PRs from forks
```

Comment:
I think this needs to be inside ${{ }}

Reply:
https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idif

Reply:
That's JS string template syntax. The output looks like:

```yaml
if: github.repository == "jhuleatt/hosting-deploy-sample"
```

Since we're outputting the exact repo name, we don't need any brackets, do we?

Reply:
Also I think it needs to be `'` single quotes around repo.

So:
```yaml
if: ${{ github.repository == 'foo/bar' }}
```

Reply:
ohh, the whole thing needs to be in brackets?

Reply:
oh yeah. [docs](https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#example-expression-in-an-if-conditional)

Reply:
per our chat, I've updated to `${{ github.event.pull_request.head.repo.full_name == github.repository }}`